pkg/yang/yangtype_test.go

```diff
@@ -0,0 +1,53 @@
+package yang
+
+import (
+	"testing"
+)
+
+func TestYangTypeEqual(t *testing.T) {
+
+	tests := []struct {
+		name      string
+		inLeft    *YangType
+		inRight   *YangType
+		wantEqual bool
+	}{{
+		name:      "both-nil",
+		inLeft:    nil,
+		inRight:   nil,
+		wantEqual: true,
+	}, {
+		name: "one-nil",
+		inLeft: &YangType{
+			Kind:           Ydecimal64,
+			FractionDigits: 5,
+		},
+		inRight:   nil,
+		wantEqual: false,
+	}, {
+		name: "name-unequal",
+		inLeft: &YangType{
+			Name:           "foo",
+			Kind:           Ydecimal64,
+			FractionDigits: 5,
+		},
+		inRight: &YangType{
+			Name:           "bar",
+			Kind:           Ydecimal64,
+			FractionDigits: 5,
+		},
+		wantEqual: true,
```

Comment:
Test the unequal case where they're the same name but different types?

Reply:
Done.